mainwindow.cc

```diff
@@ -2937,8 +2937,10 @@ void MainWindow::toggleMainWindow( bool onlyShow )
   if ( !isActiveWindow() )
   {
     qApp->setActiveWindow( this );
-    raise();
-    activateWindow();
+    if (cfg.preferences.raiseMainWindow) {
+      raise();
+      activateWindow();
+    }
     shown = true;
   }
```

Comment:
when a search action is triggered, does the mainwindow can be inactive at the same time?

Reply:
It can be visible but inactive, which I assume is what the `isActiveWindow()` call is for.

Reply:
there may be case when gd windows has invisible .at this case ,`raise()` should be called but without `activateWindow()`

Reply:
>at this case ,raise() should be called but without activateWindow()

I'm going to test this.

Reply:
No, apparently you have to disable `raise()` as well to prevent the main window from focusing itself.

Reply:
https://github.com/xiaoyifang/goldendict/blob/31c72dacdc5943ccbd4078578f4663d332c8dc52/mainwindow.cc#L2916-L2923

I think when mainwindow is invisible ,the application has choosen this branch.

Reply:
does popup dialog need to change accordingly?

Reply:
> does popup dialog need to change accordingly?

I think so.

Reply:
If the window is actually invisible, it is fine to raise it.